src/microbe_stage/editor/MicrobeEditorActionData.cs

```diff
@@ -0,0 +1,339 @@
+    /// </summary>
+    Combinable,
+}
+
+/// <summary>
+///   This is its own interface to make JSON loading dynamic type more strict
+/// </summary>
+public abstract class MicrobeEditorActionData
+{
+    /// <summary>
+    ///   Does this action cancel out with the <paramref name="other"/> action?
+    /// </summary>
+    /// <returns>
+    ///   Returns the interference mode with <paramref name="other"/>
+    /// </returns>
+    /// <para>Do not call with itself</para>
+    public abstract MicrobeActionInterferenceMode GetInterferenceModeWith(MicrobeEditorActionData other);
+
+    /// <summary>
+    ///   Combines two actions to one if possible.
+    ///   Call <see cref="MicrobeEditorActionData.GetInterferenceModeWith"/> first and check if it returns <see cref="MicrobeActionInterferenceMode.Combinable"/>
+    /// </summary>
+    /// <param name="other">The action this should be combined with</param>
+    /// <returns>Returns the combined action</returns>
+    /// <exception cref="NotSupportedException">Thrown when combination is not possible</exception>
+    public MicrobeEditorActionData Combine(MicrobeEditorActionData other)
+    {
+        if (GetInterferenceModeWith(other) != MicrobeActionInterferenceMode.Combinable)
+            throw new NotSupportedException();
+
+        return CombineGuaranteed(other);
+    }
+
+    public abstract int CalculateCost();
+
+    /// <summary>
+    ///   Combines two actions to one
+    /// </summary>
+    /// <param name="other">The action this should be combined with. Guaranteed to be combinable</param>
+    /// <returns>Returns the combined action</returns>
+    protected virtual MicrobeEditorActionData CombineGuaranteed(MicrobeEditorActionData other)
+    {
+        throw new NotImplementedException();
+    }
+}
+
+[JSONAlwaysDynamicType]
+public class PlacementActionData : MicrobeEditorActionData
+{
+    public List<OrganelleTemplate> ReplacedCytoplasm;
+    public OrganelleTemplate Organelle;
+    public Hex Location;
+    public int Orientation;
+
+    public PlacementActionData(OrganelleTemplate organelle, Hex location, int orientation)
+    {
+        Organelle = organelle;
+        Location = location;
+        Orientation = orientation;
+    }
+
+    public override MicrobeActionInterferenceMode GetInterferenceModeWith(MicrobeEditorActionData other)
+    {
+        // If this organelle got removed in this session
+        if (other is RemoveActionData removeActionData && removeActionData.Organelle.Definition == Organelle.Definition)
+        {
+            if (removeActionData.Location == Location)
+                return MicrobeActionInterferenceMode.CancelsOut;
+
+            return MicrobeActionInterferenceMode.Combinable;
+        }
+
+        return MicrobeActionInterferenceMode.NoInterference;
+    }
+
+    public override int CalculateCost()
+    {
+        return Organelle.Definition.MPCost - (ReplacedCytoplasm?.Sum(p => p.Definition.MPCost) ?? 0);
```

Comment:
Well, in master, replacing cytoplasm with metabolosomes costs 45, but here it costs 45 - 22 = 23.
Then if you replace 3 cytoplasms with a chloroplast it costs 50 - 3 * 22 = -16...

```suggestion
        return Organelle.Definition.MPCost;
```

Reply:
Wait, really? Huh, seems like I've never noticed. But this proposed fix does not work because it would break MP calculation when undoing